src/app.js

```diff
@@ -1,12 +1,88 @@
+import axios from "axios";
+import distance from "euclidean-distance";
+
+// Get the API token
+async function getToken() {
+  let token;
+  try {
+    const response = await axios.post(
+      "https://blue-bottle-api-test.herokuapp.com/v1/tokens",
+      {
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+      }
+    );
+    token = response.data.token;
+  } catch (e) {
+    console.error(`Exception occurred while fetching the token: ${e}`);
+  }
+  return token;
+}
+
+// Get all shop data from the API
+async function getShopData() {
+  const token = await getToken();
+
+  try {
+    const response = await axios.get(
+      `https://blue-bottle-api-test.herokuapp.com/v1/coffee_shops?token=${token}`,
+      {
+        headers: {
+          Accept: "application/json",
+        },
+      }
+    );
+    return response.data;
+  } catch (e) {
+    if (e.response.status == 401 || 406 || 503 || 504) {
+      console.error(`Exception occurred while fetching shop list : ${e}`);
+      return [];
+    }
+  }
+}
+
 /**
  * @param {Object} position
  * @param {Number} position.x
  * @param {Number} position.y
- * 
+ *
  * @returns {Array<position>}
  */
-export function getNearestShops(position) {
-  // code
-  
-  return [];
+
+// Calculate distance of each shop from the given coordinate
+export default async function getNearestShops(position) {
+  const shopData = await getShopData();
+
+  if (shopData.length > 0) {
+    let distancelist = [];
+
+    shopData.forEach((shop) => {
+      const calculatedDistance = distance(
+        [position.x, position.y],
+        [shop.x, shop.y]
+      );
+
+      const shopDistance = {
+        // Round distances to four decimal places
+        distance: calculatedDistance.toFixed(4),
+        shop: shop.name,
+      };
+
+      distancelist.push(shopDistance);
+    });
+
+    // Sort distanceList by distance
```

Comment:
Can you tell me when do you use comments and why? What are some of the best practices you follow related to comments.

Reply:
I try to add comments to my code when I think it might be useful for the reader. I know that adding unnecessary comments where the code explains itself is not a good practice. I am trying to improve myself on this. I added a comment here because I thought that sorting the array by the distance might be not be understood at first sight here. Until now I could not find time to read it but I am planning to read "Clean Code" book by Martin Robert to improve myself about writing cleaner code.